Fix faulty return value conversion for socket io

v-atamanenko opened the pull request.

v-atamanenko committed:
Fix faulty return value conversion for socket io

v-atamanenko committed:
fix __vita_fd_drop

isage commented on newlib/libc/sys/vita/syscalls.c:
```diff
@@ -42,8 +43,13 @@ _write_r(struct _reent * reent, int fd, const void *buf, size_t nbytes)
-			ret = __vita_glue_socket_send(fdmap->sce_uid, buf, nbytes, 0);
+		ret = sceNetSend(fdmap->sce_uid, buf, nbytes, 0);
+		if (ret < 0) {
+			if (ret != -1 )
```
Is that needed?

isage commented on newlib/libc/sys/vita/syscalls.c:
```diff
@@ -265,8 +271,13 @@ _read_r(struct _reent *reent, int fd, void *ptr, size_t len)
-			ret = __vita_glue_socket_recv(fdmap->sce_uid, ptr, len, 0);
+		ret = sceNetRecv(fdmap->sce_uid, ptr, len, 0);
+		if (ret < 0) {
+			if (ret != -1 )
```
Same

isage commented:
Remove glue funcs too if they aren't used anymore?

v-atamanenko committed:
Drop now unused vitaglue functions; drop redundant '!= -1' condition

v-atamanenko committed:
revert accidental whitespace changes

isage commented:
Looks good.

v-atamanenko commented:
tested, works well

v-atamanenko commented:
To the merging person: please add @isage in Co-authored-by: in squash commit

frangarcj commented:
@d3m3vilurr you did the original code, can you take a look?

I see some lw and fd drop calls that are new

v-atamanenko commented:
these are not really new, just moved from either now-removed nested vitaglue function, or copied from the bottom of the function (since we need to return earlier). the behavior isn't changed in this sense.

d3m3vilurr commented on newlib/libc/sys/vita/socket.c:
```diff
@@ -467,22 +467,4 @@ int	socket(int domain, int type, int protocol)
 	return s;
 }
-
-int __vita_glue_socket_close(SceUID scefd)
```
personally i want to keep this week reference.

d3m3vilurr reviewed (commented):
i cannot find any changes of this PR.
it seems just remove week reference isn't it?

could you explain actual changing part?

v-atamanenko commented:
@d3m3vilurr previously, sceErrNo got converted in vitaglue functions and returned from there back to syscalls. This caused the behavior I experienced: when trying to read() something bigger then the buffer, read should return -1 and EAGAIN in errno. But in this case it returned 11 and errno 0, as if there was 11 bytes read. This PR fixes read, write, and close which are all prone to this error.

You can read even more detailed story in the henkaku discord, we discussed the issue there with isage.
https://discord.com/channels/439481392548675594/465960127963463680/897881471434522734

On the side note, what’s the point of a weak ref here?
